### PR TITLE
Google and Github OAuth

### PR DESCRIPTION
**Pull request by jonas-kgomo**
Login for google and github users , accounts should be different.

### CODE REVIEW

**Comment by chalkedgoose** [comment]
Is there any chance we could get a preview of what this PR does a Gif or MP4?

**Comment by coderinblack08** [comment]
He sent a mp4 to me, works great

**Comment by coderinblack08** [comment]
Lol why did you modify the footer and add `@supabase/ui`, seems totally unrelated

**Comment by jonas-kgomo** [comment]
after our discussion about using our design system I removed superbase-ui.  About footer,  i noticed the footer wasn't showing in `develop` branch on `components/layout`, so i added that as well (after @chalkedgoose
suggested)

**Review by chalkedgoose — APPROVED** [review]

File: pages/index.tsx [hunk 1]
```diff
@@ -30,8 +33,14 @@ const Index: React.FC = () => {
           </Button>
         </div>
         <div className="grid grid-cols-2 items-center gap-2 mt-5">
-          <Button>Or Join with Google</Button>
-          <Button>Or Join with GitHub</Button>
+          <Button
+            onClick={async () =>
+            { await supabase.auth.signIn({ provider: 'google' }) }}
```
Comment (chalkedgoose) [inline pages/index.tsx:38]: Is there a reason for using async await in your onClick callback?
wont the browser execute the promise anyways?

File: pages/index.tsx [hunk 2]
```diff
@@ -30,8 +33,14 @@ const Index: React.FC = () => {
...
+          > Join with Google</Button>
+          <Button
+           onClick={async () =>
```
Comment (chalkedgoose) [inline pages/index.tsx:41]: Same here.

**Review by chalkedgoose — CHANGES_REQUESTED** [review]

File: pages/index.tsx (hunk 1 above)
Comment (jonas-kgomo) [reply pages/index.tsx:38]: Correct,  `await` makes sense for `signout`, not sign in, I will remove this

**Comment by chalkedgoose** [comment]
Rebasing your PR should get rid of conflict

**Pull request merged**